src/ai/backend/manager/models/vfolder.py

```diff
@@ -1213,6 +1234,63 @@ async def ensure_quota_scope_accessible_by_user(
     raise InvalidAPIParameters
 
 
+async def get_sessions_by_mounted_folder(
+    db_session: SASession, vfolder_id: VFolderID
+) -> tuple[uuid.UUID]:
+    """
+    Return a tuple of sessions.id that the give folder is mounted on.
+    """
+
+    @attrs.define(slots=True)
+    class SimpleVFolderMount(JSONSerializableMixin):
+        vfid: VFolderID
+
+        def to_json(self) -> dict[str, Any]:
+            return {
+                "vfid": str(self.vfid),
+            }
+
+        @classmethod
+        def from_json(cls, obj: Mapping[str, Any]) -> SimpleVFolderMount:
+            return cls(**cls.as_trafaret().check(obj))
+
+        @classmethod
+        def as_trafaret(cls) -> t.Trafaret:
+            return t.Dict({
+                t.Key("vfid"): tx.VFolderID,
+            })
+
+    class SimpleSessionRow(Base):
```

Comment:
Is this kind of specialization necessary to "minimize the queried columns"?
Could you compare the actual stringification results of the SQLAlchemy query object when using this class or the original class?

If you have added this inner class due to circular import issues, you could just import the session model classes inside the function.

Reply:
I left comments.
It is because of the query condition **vfolder_mounts.contains([VFolderMount(vfid=vfolder_id)])**.
`VFolderMount` requires all name, vfid, vfsubpath, host_path etc fields to initialize and it is impossible to query with a condition like this **vfolder_mounts.contains([{"vfid": vfolder_id)])**.
So I just made a simplified type and an ORM class for this query.

Reply:
So, the problem is that we cannot specify a partially initialized `VFolderMount` object for a SELECT context. (When INSERT-ing, we need all fields and our `JSONSerializableMixin` expects this.)

Could we find a way to detect the query context when `process_bind_param()` is called and consider introduction of a new optional flag `allow_partial=True|False` to `JSONSerializableMixin.to_json()`?

Reply:
As you've researched, it seems that it's not trivial to detect the query type inside type decorators.
Let's conclude this design as-is: adding a partial declaration of `SessionRow` as `SimpleSessionRow` to allow raw JSONB column queries.